components/automate-cli/cmd/chef-automate/start_linx.go

```diff
@@ -9,59 +10,293 @@ import (
-			return status.Annotate(err, status.HabCommandError)
+	}
+	if startCmdFlags.opensearch {
+		if isManagedServicesOn() {
+			return status.Errorf(status.InvalidCommandArgsError, errorOnManagedServicesStart, "OpenSearch")
 		}
-	} else {
-		systemctlCmd := exec.Command("systemctl", "start", "chef-automate.service")
-		systemctlCmd.Stdout = os.Stdout
-		systemctlCmd.Stderr = os.Stderr
-		if err := systemctlCmd.Run(); err != nil {
-			return status.Annotate(err, status.ServiceStartError)
+		backendIps := infra.Outputs.OpensearchPrivateIps.Value
+		err = checkNodes(args, sshUtil, backendIps, opensearchService, writer)
+		if err != nil {
+			errorList.PushBack(err.Error())
+		}
+	}
+	if startCmdFlags.postgresql {
+		if isManagedServicesOn() {
+			return status.Errorf(status.InvalidCommandArgsError, errorOnManagedServicesStart, "Postgresql")
+		}
+		backendIps := infra.Outputs.PostgresqlPrivateIps.Value
+		err = checkNodes(args, sshUtil, backendIps, postgresqlService, writer)
+		if err != nil {
+			errorList.PushBack(err.Error())
 		}
 	}
+	if errorList.Len() > 0 {
+		return status.New(status.ServiceStartError, getSingleErrorFromList(errorList).Error())
+	}
+	return nil
+}
 
+func runStartStandalone() error {
+	writer.Title("Starting Chef Automate")
+	systemctlCmd := exec.Command("systemctl", "start", "chef-automate.service")
+	systemctlCmd.Stdout = os.Stdout
+	systemctlCmd.Stderr = os.Stderr
+	if err := systemctlCmd.Run(); err != nil {
+		return status.Annotate(err, status.ServiceStartError)
+	}
 	return nil
 }
 
-func init() {
-	RootCmd.AddCommand(startCommand)
+func runStartDevMode() error {
+	writer.Title("Starting Chef Automate")
+	if err := exec.Command("hab", "sup", "status").Run(); err == nil {
+		return nil
+	}
+
+	if err := os.MkdirAll("/hab/sup/default", 0755); err != nil {
+		return status.Annotate(err, status.FileAccessError)
+	}
+	writer.Title("Launching the Habitat Supervisor in the background...")
+	out, err := os.Create("/hab/sup/default/sup.log")
+	if err != nil {
+		return status.Annotate(err, status.FileAccessError)
+	}
+	startSupCmd := exec.Command("hab", "sup", "run")
+	startSupCmd.Env = os.Environ()
+	startSupCmd.Env = append(startSupCmd.Env,
+		"DS_DEV=true",
+		"CHEF_AUTOMATE_SKIP_SYSTEMD=true",
+		"HAB_LICENSE=accept-no-persist",
+	)
+	startSupCmd.Stdout = out
+	startSupCmd.Stderr = out
+	startSupCmd.SysProcAttr = &syscall.SysProcAttr{
+		Setpgid: true,
+	}
+	if err := startSupCmd.Start(); err != nil {
+		return status.Annotate(err, status.HabCommandError)
+	}
+	return nil
+}
+
+func checkNodes(args []string, sshUtil SSHUtil, ips []string, remoteService string, cliWriter *cli.Writer) error {
+	if len(ips) == 0 {
+		writer.Errorf("No %s IPs are found", remoteService)
+		return status.Errorf(1, "No %s IPs are found", remoteService)
+	}
+	if remoteService == opensearchService || remoteService == postgresqlService {
+		return startBackEndNodes(args, sshUtil, ips, remoteService, cliWriter)
```

Comment:
Making checkNodes testable:

```suggestion
		return startBackEndNodes(args, sshUtilMap, ips, remoteService, cliWriter)
```